overlord/snapshotstate/backend/backend.go

```diff
@@ -157,6 +157,63 @@ func Filename(snapshot *client.Snapshot) string {
 	return filepath.Join(dirs.SnapshotsDir, fmt.Sprintf("%d_%s_%s_%s.zip", snapshot.SetID, snapshot.Snap, snapshot.Version, snapshot.Revision))
 }
 
+// EstimateSnapshotSize calculates estimated size of the snapshot.
+// XXX: this doesn't take users into account yet, because it is used only
+// for automatic snapshots at the moment.
+func EstimateSnapshotSize(si *snap.Info) (int64, error) {
+	var total int64
+	calculateSize := func(path string, finfo os.FileInfo, err error) error {
+		if finfo.Mode().IsRegular() {
```

Comment:
Can we have directories here? They may use a variable number of filesystem blocks, but unsure how Size() behaves in this case (the documentation doesn't say much about non-regular files :\ )

Reply:
IsRegular() check will only consider file sizes here.

And yes, directories are reported by filepath.Walk as well; I'm not sure it makes sense to take their blocks into account because 1) it's an estimate only (and a pessimistic one, no compression effects taken into account); 2) compressed snapshot most likely uses completely different organization of directories.

Reply:
Right, it's an estimate so there's no point in trying to be precise with directory sizes.